common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/FinalizeShuffleMerge.java

```diff
@@ -30,12 +30,15 @@
  */
 public class FinalizeShuffleMerge extends BlockTransferMessage {
   public final String appId;
+  public final int attemptId;
```

Comment:
FinalizeShuffleMessage is sent from the driver to the shuffle services. Is it possible for the driver of a previous attempt to be alive when the driver of a new attempt is running? 
I think whenever the server receives `FinalizeShuffeMerge` it should always assume it is from the latest attempt

Reply:
In rare cases, the driver container from the last attempt may still be running in the cluster whereas the driver from the second attempt has been launched. It is better that we can also have the attemptId in the FinalizeShuffleMessage.